### PR TITLE
[BE] add import profile feature

### PR DESCRIPTION
**Pull request by LuiggiPasacheL**
# Add import profile feature
1. First, This feature is made with the **dto** pattern due to the complexity of the request and the response data.
2. The request is converted to a **domain profile** to be proccesed into the application use case.
3. This use case returns the **domain profile** and this profile is send to response.
4. This response converts the domain profile to a **activityPub Person**, and I use this library to marshall **json response**.
---
Add feature import-profile following the hexagonal pattern:
This feature receives a json according to the specifications of #55 and updates the user in the PUT /profile/import path it receives:
```json
{
	"@context": [
		"https://www.w3.org/ns/activitystreams",
		{
			"@language": "es"
		}
	],
	"type": "Profile",
	"actor": "https://appsocial.com/sofia/",
	"name": "Perfil de Sofia",
	"object": {
		"id": "https://appsocial.com/sofia/id1234",
		"type": "Person",
		"name": "Sofia Rodriguez",
		"preferredUsername": "SofiR",
		"summary": "Amante de la naturaleza y entusiasta de la tecnología.",
		"profileImage": "https://appsocial.com/sofia/profileImage.jpg",
		"coverImage": "https://appsocial.com/sofia/coverImage.jpg",
		"endpoints": {
			"sharedInbox": "https://appsocial.com/inbox"
		},
		"importedFrom": "https://mastodon.ejemplo.com/@sofiR"
	},
	"to": [
		"https://appsocial.com/usuarios/"
	],
	"cc": "https://appsocial.com/seguidores/sofia"
}
```
If the JSON is correct, an UPDATE will be made to the active user's profile and the following will be returned (based on the previous example):
```json
{
	"response": "Profile Imported Succesfully",
	"data": {
		"id": "https://appsocial.com/sofia/id1234",
		"type": "Person",
		"name": "Sofia Rodriguez",
		"summary": "Amante de la naturaleza y entusiasta de la tecnología.",
		"icon": "https://kenzoishii.example.com/image/165987aklre4",
		"preferredUsername": "SofiR",
		"endpoints": {
			"sharedInbox": "https://appsocial.com/inbox"
		}
	}
}
```

### CODE REVIEW

File: backend/internal/profile/infrastructure/routes/profileHandler.go [hunk 1]
```diff
@@ -0,0 +1,25 @@
+package infrastructure_routes
+
+import (
+	"net/http"
+
+	"github.com/go-chi/chi"
```
Comment (ekimike) [inline backend/internal/profile/infrastructure/routes/profileHandler.go:6]: Is this library being used? If not, you may remove it.

File: backend/internal/profile/test/infrastructure/dto/importProfileResponse_test.go [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+        Response: "Profile Imported Succesfully",
+        Data: expectedPerson,
+    }
+
+	// Crear una nueva respuesta ImportProfileResponse
+	response := dto.NewImportProfileResponse(expectedPerson)
+
+	// Verificar que los campos de la respuesta sean los esperados
+	if response.Response != "Profile Imported Succesfully" {
+		t.Errorf("Se esperaba el valor 'Profile Imported Succesfully', pero se obtuvo '%s'", response.Response)
```
Comment (ekimike) [inline backend/internal/profile/test/infrastructure/dto/importProfileResponse_test.go:31]: Shall we use English here?

File: backend/internal/profile/test/infrastructure/dto/importProfileResponse_test.go [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+	}
+
+    if expectedResponse.Data != response.Data {
+        t.Errorf("Se esperaba la respuesta '%s', pero se obtuvo '%s'", expectedResponse, response)
```
Comment (ekimike) [inline backend/internal/profile/test/infrastructure/dto/importProfileResponse_test.go:35]: Shall we use English here?

File: backend/internal/profile/test/infrastructure/importProfileHandler_test.go [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+    }`)
+
+	req := httptest.NewRequest("POST", "/import-profile", bytes.NewBuffer(requestBody))
+	req.Header.Set("Content-Type", "application/json")
+	res := httptest.NewRecorder()
+
+	handler.ImportProfile(res, req)
+
+	if res.Code != http.StatusCreated {
+		t.Errorf("Se esperaba el código de estado %d, pero se obtuvo %d", http.StatusCreated, res.Code)
```
Comment (ekimike) [inline backend/internal/profile/test/infrastructure/importProfileHandler_test.go:57]: Shall we use English?

File: backend/vendor/git.sr.ht/~mariusor/go-xsd-duration/LICENSE
Comment (ekimike) [inline backend/vendor/git.sr.ht/~mariusor/go-xsd-duration/LICENSE:1]: Do weed to ignore this?

File: backend/vendor/git.sr.ht/~mariusor/go-xsd-duration/xsd_duration.go
Comment (ekimike) [inline backend/vendor/git.sr.ht/~mariusor/go-xsd-duration/xsd_duration.go:1]: Do we need to ignore this thru `.gitignore`?

File: backend/vendor/github.com/go-ap/activitypub/.build.yml
Comment (ekimike) [inline backend/vendor/github.com/go-ap/activitypub/.build.yml:1]: Do we need to ignore this thru `.gitignore`?

File: backend/vendor/github.com/go-ap/activitypub/.gitignore
Comment (ekimike) [inline backend/vendor/github.com/go-ap/activitypub/.gitignore:1]: Do we need to ignore this thru `.gitignore`?

File: backend/vendor/github.com/go-ap/activitypub/LICENSE
Comment (ekimike) [inline backend/vendor/github.com/go-ap/activitypub/LICENSE:1]: Do we need to ignore this thru `.gitignore`?

File: backend/vendor/github.com/go-ap/activitypub/Makefile
Comment (ekimike) [inline backend/vendor/github.com/go-ap/activitypub/Makefile:1]: Do we need to ignore this thru `.gitignore`?

File: backend/vendor/github.com/go-ap/activitypub/README.md
Comment (ekimike) [inline backend/vendor/github.com/go-ap/activitypub/README.md:1]: Do we need to ignore this thru `.gitignore`?

File: backend/vendor/github.com/go-ap/activitypub/activity.go
Comment (ekimike) [inline backend/vendor/github.com/go-ap/activitypub/activity.go:1]: Do we need to ignore this thru `.gitignore`?

File: backend/vendor/github.com/go-ap/activitypub/actor.go
Comment (ekimike) [inline backend/vendor/github.com/go-ap/activitypub/actor.go:1]: Do we need to ignore this thru `.gitignore`?

**Review by ekimike — COMMENTED** [review]
There are a lot of files coming from `backend/vendor` which seems should NOT be there, please confirm

File: backend/internal/profile/infrastructure/import/importProfileHandler.go [hunk 5]
```diff
@@ -0,0 +1,45 @@
+package infrastructure_import
```
[the remark this answers is not in the document]
Comment (LuiggiPasacheL) [reply backend/internal/profile/infrastructure/import/importProfileHandler.go:1]: Yes, I'm working on it

File: backend/go.sum
[the remark this answers is not in the document]
Comment (LuiggiPasacheL) [reply backend/go.sum:1]: After what was agreed with the backend team I am going to delete and ignore that folder

File: backend/internal/profile/test/infrastructure/dto/importProfileResponse_test.go (hunk 2 above)
Comment (LuiggiPasacheL) [reply backend/internal/profile/test/infrastructure/dto/importProfileResponse_test.go:31]: Yes, I made a mistake in the language

**Review by miguel1808 — APPROVED** [review]
The profile import functionality seems to be well implemented and follows the hex pattern we are using. It is also important that you have followed the specifications in [#55](https://github.com/unmsmfisi-socialapplication/social_app/issues/55) and provided detailed examples of JSON requests and responses. This will be very helpful for documentation and testing.

**Pull request merged**